executor/analyze.go

```diff
@@ -206,9 +211,70 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			}
 		}
 	}
+	err = e.saveAnalyzeOptsV2()
+	if err != nil {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+	}
 	return statsHandle.Update(e.ctx.GetInfoSchema().(infoschema.InfoSchema))
 }
 
+func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
+	if !variable.PersistAnalyzeOptions.Load() || e.V2AnalyzeOptions == nil {
+		return nil
+	}
+	err := saveAnalyzeOpts(e.V2AnalyzeOptions, e.ctx.(sqlexec.RestrictedSQLExecutor))
```

Comment:
It seems `e.ctx` is the same `sessionctx.Context` under which we execute `analyze`. Will there be any problem if we execute some SQLs when the execution of `analyze` is halfway through with the same `sessionctx.Context`? For other SQLs we want to execute in `(*AnalyzeExec).Next`, for example, the SQLs to dump the collected statistics to disk,
https://github.com/pingcap/tidb/blob/9ad00962f6561b25fe0fe5290ba16de1c3901ff7/executor/analyze.go#L174 
we use `Handle.mu.ctx` as `sessionctx.Context`.

Reply:
It calls RestrictedSQLExecutor.ExecRestrictedStmt which uses session from sysSessionPool().Get() to execute this sql. I can see calculateEstimateSampleStep works the same way. And I think it's not must to lock and use Handle.mu.ctx.